core/local/steps/incomplete_fixer.js

```diff
@@ -0,0 +1,95 @@
+/* @flow */
+
+const path = require('path')
+
+const { id } = require('../../metadata')
+const stater = require('../stater')
+
+// Drop incomplete events after this delay (in milliseconds).
+// TODO tweak the value (the initial value was chosen because it looks like a
+//      good value, it is not something that was computed)
+const DELAY = 3000
+
+// TODO add unit tests and logs
+
+/*::
+import type Buffer from './buffer'
+import type { AtomWatcherEvent } from './event'
+import type { Checksumer } from '../checksumer'
+
+type IncompleteItem = {
+  event: AtomWatcherEvent,
+  timestamp: number,
+}
+*/
+
+// When a file is added or updated, and it is moved just after, the first event
+// is marked as incomplete by addChecksum because we cannot compute the
+// checksum at the given path. But the event is still relevant, in particular if
+// a directory that is an ancestor of this file has been moved. With the renamed
+// event, by comparing the path, we can extrapolate the new path and check with
+// fs.stats if we have a file here.
+//
+// Cf test/property/local_watcher/swedish_krona.json
+module.exports = function (buffer /*: Buffer */, opts /*: { syncPath: string , checksumer: Checksumer } */) /*: Buffer */ {
+  const incompletes = []
+  return buffer.asyncMap(async (events) => {
+    // Filter out the incomplete events
+    const batch = []
```

Comment:
Maybe extract `async function fixIncompleteEvents(incompletes, events, opts)` to make it easier to test out of `buffer.asyncMap()`?

Reply:
To be done when adding unit tests, same as https://github.com/cozy-labs/cozy-desktop/pull/1369#discussion_r245349730